store/rootmulti/store.go

```diff
@@ -441,9 +443,15 @@ func (rs *Store) Query(req abci.RequestQuery) abci.ResponseQuery {
 		return sdkerrors.QueryResult(sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "proof is unexpectedly empty; ensure height has not been pruned"))
 	}
 
-	commitInfo, errMsg := getCommitInfo(rs.db, res.Height)
-	if errMsg != nil {
-		return sdkerrors.QueryResult(err)
+	var commitInfo commitInfo
```

Comment:
I'd add a comment here to explain this block:

```go
// If the request's height is the latest height we've committed, then utilize the store's lastCommitInfo
// as this commit info may not be flushed to disk. Otherwise, we query for the commit info from disk.
```